MMDevice/MMDevice.h

```diff
@@ -1428,4 +1445,307 @@ namespace MM {
       MM_DEPRECATED(virtual void ClearPostedErrors(void)) = 0;
    };
 
+   /**
+    * \brief   Device interface for managing multi-dimensional datasets in storage
+    * \details The Storage class provides an interface for creating, managing, and accessing
+    *          multi-dimensional datasets. It supports operations such as creating new datasets,
+    *          adding images, retrieving metadata, and managing dataset dimensions and coordinates.
+    *
+    * \note Implementation Notes:
+    *       - All metadata variables are ASCII strings, typically JSON encoded, but not required
+    *       - Functions returning metadata allocate buffers on the heap that must be released by the caller.
+    *       - There are no explicit restrictions to image size, number of dimensions, size of dimensions and size of metadata
+    *       - The same device may support opening of multiple datasets simultaneously, or only one dataset at a time.
+    *       - The API allows for random access in image insertions and image retrieval, but the implementation may not support it.
+    *       - Most formats allow only appending images in the order of dimensions.
+    *       - Allowing random access to images (during retrievel) is recommended.
+    *       - Lazy loading of pixel data is recommended for large datasets.
+    *
+    * \inherit Device
+    */
+   class Storage : public Device {
+   public:
+      Storage() {}
+      virtual ~Storage() {}
+
+      virtual DeviceType GetType() const { return Type; }
+      static const DeviceType Type;
+
+      /**
+       * \brief   Creates a new dataset
+       *
+       * \param   path The path where the dataset will be created (parent directory)
+       * \param   name The name of the dataset (may be modified by the implementation to avoid overwriting existing datasets)
+       * \param   numberOfDimensions Number of dimensions in the dataset
+       * \param   shape Array defining the size of each dimension
+       * \param   pixType The data type for pixel storage
+       * \param   meta Dataset metadata string
+       * \param   [out] handle Output parameter for the dataset handle
+       * \return  Status code indicating success or failure
+       *
+       * \note    - Dimensions are ordered from slowest changing to fastest changing.
+       *          - Typically we are storing images and the last two dimensions are always image height (Y) and width (X)
+       *          - Some implementation may allow non-image data to be stored
+       *          - Declared size for the slowest (first) dimension can be exceeded during acquisition
+       */
+      virtual int Create(const char* path, const char* name, int numberOfDimensions, const int shape[],
+                         MM::StorageDataType pixType, const char* meta, char* handle) = 0;
```

Comment:
Is there an important reason to use a string handle? Wondering if it might be simpler to use a `uint64_t` handle if the value itself has no meaning.

Reply:
Also, for the metadata parameter, I would suggest using a pointer and size (`std::size_t`) rather than a null-terminated string. This would not only eliminate a scan on the receiving side to determine the length, but would also allow metadata that contains internal NUL bytes (not that we'd recommend that).

This applies to all passing of metadata in and out, of course.